### PR TITLE
has_any_layout utility function

### PR DESCRIPTION
**Pull request by cpuhrsch**
### Description
This is a proposal with a few integration sites for a utility function that can help us save a few lines of code and typos.

### Testing
This is a refactor and covered by existing tests

### CODE REVIEW

File: aten/src/ATen/native/sparse/SparseCsrTensorMath.h [hunk 1]
```diff
@@ -11,9 +11,7 @@ namespace impl {
 // Returns true if all entries of self are zero
 // TODO: This has potential to be a generic helper
 inline bool _is_sparse_and_zero(const Tensor& self) {
-  if (self.layout() == kSparse || self.layout() == kSparseCsr ||
-      self.layout() == kSparseCsc || self.layout() == kSparseBsr ||
-      self.layout() == kSparseBsc) {
+  if (has_any_layout(self.layout(), kSparse, kSparseCsr, kSparseCsc, kSparseBsr, kSparseBsc)) {
```
Comment (pearu) [inline aten/src/ATen/native/sparse/SparseCsrTensorMath.h:14]: This should read:
```suggestion
  if (has_any_layout(self, kSparse, kSparseCsr, kSparseCsc, kSparseBsr, kSparseBsc)) {
```

**Review by pearu — CHANGES_REQUESTED** [review]
Other than one bug, this looks good to me.

Would it make sense to expose `has_any_layout` as a tensor method?

**Comment by cpuhrsch** [comment]
@pearu - Maybe, but then we'll potentially need to discuss it from an API perspective. Right now it's just an innocent refactor that saves us time and bugs.

**Comment by bhosmer** [comment]
see [here](https://github.com/pytorch/pytorch/pull/82122#discussion_r938993325) for the argument against 😁 (and an alternative)

**Pull request closed without merging**